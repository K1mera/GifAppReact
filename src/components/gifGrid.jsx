import { GifItem } from "./GifItem"
import { useFetchGifs } from "../hooks/useFetchGifs"


export const GifGrid = ({ category }) => {

    const { images, loading } = useFetchGifs(category)

    // const [images, setImages] = useState([])
    // const getImg = async () => {
    //     const newImg = await getGifs( category );
    //     setImages(newImg)
    // }

    // console.log()

    // useEffect(() => {
    //     getImg();
    // }, [])


    return (
        <>
            <h3>{category}</h3>
            {
                loading && (<h2>Loading...</h2>)
            }
            <div className="card-grid">
                {
                    images.map(((img) =>
                        <GifItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }

            </div>

        </>
    )
}
