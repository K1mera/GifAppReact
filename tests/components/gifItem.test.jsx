import { render, screen } from '@testing-library/react'
import { GifItem } from "../../src/components/GifItem"




describe('gifItem test', () => { 

    const title = 'Menudo title'
    const url = 'https://melasuda/todo.jpg'

    test('should match the snapshot', () => { 
        const { container } = render(<GifItem title={ title } url={ url } /> )
        expect( container ).toMatchSnapshot();
     })

    test('should return the img src and the alt attributes', () => {
        render(<GifItem title={ title } url={url} />);
        // expect(screen.getByRole('img').src).toBe( url );
        // expect(screen.getByRole('img').alt).toBe( title );
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(src);
        expect(alt).toBe(alt);  
    });

    test('should return the proper title given as prop', () => { 
        render(<GifItem title={ title } url={ url }/>);
        expect( screen.getByText( title )).toBeTruthy;
     });
     
 })

