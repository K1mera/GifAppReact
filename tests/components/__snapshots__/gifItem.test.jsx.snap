// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gifItem test should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Menudo title"
      src="https://melasuda/todo.jpg"
    />
    <p>
      Menudo title
    </p>
  </div>
</div>
`;
